docs/examples/caffe_build_index.md

```diff
@@ -0,0 +1,54 @@
+The following is a concrete example of performing nearest neiborhood computation
+using a set of ten butterfly images. It assumes you have set up the
+caffe (vrc2) environment (http://caffe.berkeleyvision.org/install_apt.html)
+set correctly.
+
+Run following scripts to generate image_mean_filepath, network_model_filepath,
+and network_prototxt_filepath
+
+> caffe_src/ilsvrc12/get_ilsvrc_aux.sh
+> carffe_src/scripts/download_model_binary.py  ./models/bvlc_reference_caffenet/
+
+```python
+from smqtk.algorithms.nn_index.flann import FlannNearestNeighborsIndex
+
+# Import some butterfly data
+urls = ["http://www.comp.leeds.ac.uk/scs6jwks/dataset/leedsbutterfly/examples/{:03d}.jpg".format(i) for i in range(1,11)]
+from smqtk.representation.data_element.url_element import DataUrlElement
+el = [DataUrlElement(d) for d in urls]
+
+# Create a model. This assumes you have set up the colordescriptor executable.
+from smqtk.algorithms.descriptor_generator import get_descriptor_generator_impls
+cd = get_descriptor_generator_impls()['CaffeDescriptorGenerator'](
+        network_prototxt_filepath="caffe/models/bvlc_reference_caffenet/deploy.prototxt",
+        network_model_filepath="caffe/models/bvlc_reference_caffenet/bvlc_reference_caffenet.caffemodel",
+        image_mean_filepath="caffe/data/ilsvrc12/imagenet_mean.binaryproto",
+        return_layer="fc7",
+        batch_size=1,
+        use_gpu=False,
+        gpu_device_id=0,
+        network_is_bgr=True,
+        data_layer="data",
+        load_truncated_images=True)
+# cd.generate_model(el)
```

Comment:
going to want to remove this commented-out line in relation with the specific executable being used. It might be confusing to people.